Assets/Scripts/Injector.cs

```diff
@@ -0,0 +1,36 @@
+﻿using DELTation.DIFramework;
+using DELTation.DIFramework.Containers;
+using UnityEngine;
+
+public sealed class Injector : DependencyContainerBase
+{
+    [SerializeField] private BallView _ballView;
+    [SerializeField] private ManualView _manualView;
+    [SerializeField] private bool _barToggle;
+
+    public GameObject Bar;
+    public GameObject Button;
+
+    protected override void ComposeDependencies(ICanRegisterContainerBuilder builder)
+    {
+        if (_barToggle)
+        {
+            Bar.SetActive(true);
+            Button.SetActive(true);
+            builder.Register(_manualView);
+        }
+        else
+        {
+            Bar.SetActive(false);
+            Button.SetActive(false);
+            builder.Register(_ballView);
+        }
+
+        builder
+            .Register(ScriptableObject.CreateInstance<DummyCfg>())
```

Comment:
this should be an asset in your project folder, but is okay for the purposes of the homework